### PR TITLE
feat: Support `#[stack_trace]` op2 arg

### PR DESCRIPTION
**Pull request by bartlomieju**
Needed for https://github.com/denoland/deno/pull/25502

### CODE REVIEW

File: ops/op2/dispatch_slow.rs [hunk 1]
```diff
@@ -183,6 +192,22 @@ pub(crate) fn with_scope(generator_state: &mut GeneratorState) -> TokenStream {
...
 
+pub(crate) fn with_stack_trace(
+  generator_state: &mut GeneratorState,
+) -> TokenStream {
+  generator_state.needs_opctx = true;
+  gs_quote!(generator_state(stack_trace, opctx, scope) =>
+    (let #stack_trace = if #opctx.enable_stack_trace_arg {
+      let hs = &mut v8::HandleScope::new(&mut #scope);
+      // TODO(bartlomieju): use a cheaper string here, or maybe undefined?
+      let stack_trace_msg = v8::String::new(hs, "unused").unwrap();
```
Comment (bartlomieju) [inline ops/op2/dispatch_slow.rs:203]: Fix before landing

File: ops/op2/dispatch_slow.rs [hunk 2]
```diff
@@ -183,6 +192,21 @@ pub(crate) fn with_scope(generator_state: &mut GeneratorState) -> TokenStream {
...
 }
 
+pub(crate) fn with_stack_trace(
+  generator_state: &mut GeneratorState,
+) -> TokenStream {
+  generator_state.needs_opctx = true;
+  gs_quote!(generator_state(stack_trace, opctx, scope) =>
+    (let #stack_trace = if #opctx.enable_stack_trace_arg {
+      let hs = &mut deno_core::v8::HandleScope::new(&mut #scope);
+      let stack_trace_msg = deno_core::v8::String::new(hs, "").unwrap();
```
Comment (littledivy) [inline ops/op2/dispatch_slow.rs:202]:
```suggestion
      let stack_trace_msg = deno_core::v8::String::empty(hs);
```

**Review by littledivy — APPROVED** [review]
LGTM!

**Comment by devsnek** [comment]
why is this a magic op argument? can't you just call these V8 apis if you need them?

**Comment by bartlomieju** [comment]
> why is this a magic op argument? can't you just call these V8 apis if you need them?

I can, but doing so in every op that does permission check is cumbersome, especially for async ops which would need to be rewritten from `async fn ...` to `fn ...() -> impl Future` to be able to accept a handle scope. Plus I need a "global switch" to activate or deactivate it, since we only want to collect these traces if a proper env var is passed (because it's so slow).

**Pull request merged**